src/API/Site/Controllers/MerchantCenter/PriceBenchmarksController.php

```diff
@@ -83,6 +96,31 @@ protected function get_price_benchmarks_callback(): callable {
 		};
 	}
 
+	/**
+	 * Callback for the price benchmarks summary endpoint.
+	 *
+	 * @return callable
+	 */
+	protected function get_price_benchmarks_summary_callback(): callable {
+		return function ( Request $request ) {
+			try {
+				/** @var MerchantPriceBenchmarks $merchant */
+				$merchant = $this->container->get( MerchantPriceBenchmarks::class );
+
+				// Fetch benchmark and price insights data.
+				$benchmark_data      = $merchant->get_benchmark_data( $this->prepare_query_arguments( $request ) );
+				$price_insights_data = $merchant->get_price_insights( $this->prepare_query_arguments( $request ) );
+
+				// Generate summary data.
+				$summary_data = $this->get_price_benchmarks_summary( $benchmark_data, $price_insights_data );
```

Comment:
Rather than calculating the summary data in the API controller let's create a new service class at `src/MerchantCenter/PriceBenchmarks.php` and we'll use that service class to implement the functionality necessary to refresh the local DB and return the summary data to other parts of the application. This will be similar to how `src/MerchantCenter/MerchantStatuses.php` works to refresh product stats.

Reply:
When i add service class and tried to call it, it show me an error 🤔

Reply:
Looks like you had the service class registered with the `RESTServiceProvider`, which is meant for REST API controllers. I've moved it to the `CoreServiceProvider` in [ebd2def](https://github.com/woocommerce/google-listings-and-ads/pull/2857/commits/ebd2def683dfde943fb6db364b9387c278da081f) and that seemed to help. The `get_summary()` method still needs some work. It should probably use the `MerchantPriceBenchmarksQuery` class rather than accessing the `MerchantPriceBenchmarksTable` class directly.

Reply:
In IB it says that

> A new get_price_benchmarks_summary_callback() callback should use a new get_price_benchmarks_summary() method on a new \MerchantCenter\PriceBenchmarks service class which will query the local DB.

or do we rely on  `MerchantPriceBenchmarksQuery`?

Reply:
Updated in https://github.com/woocommerce/google-listings-and-ads/pull/2857/commits/7db77e71e089b150a3bc62ff69bfed345c4a6380